src/engine/interfaces.ts

```diff
@@ -0,0 +1,40 @@
+export enum RAFPhase {
+  MEASURE,
+  MUTATE
+}
+
+export interface QueueElementInterface {
```

Comment:
You might want to nix the `*Interface` since the only thing you can `implements` is an interface.